DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces_overview/host-pipes/src/vector_add.cpp

```diff
@@ -0,0 +1,116 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>  // sycl.hpp comes first to prevent compiler crash
+
+#define VECT_SIZE 256
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class IDSimpleVAddPipes;
+
+using PipeProps = decltype(sycl::ext::oneapi::experimental::properties(
+    sycl::ext::intel::experimental::ready_latency<0>));
+
+class IDPipeA;
```

Comment:
to be consistent with other code samples would it make sense to include all these at the top

Reply:
I moved all the ID to the top, I think we keep pipe declarations next to the kernel that uses them to improve readability.